pillarbox-demo/src/main/java/ch/srgssr/pillarbox/demo/service/DemoMediaLibraryService.kt

```diff
@@ -0,0 +1,155 @@
+
+    private lateinit var demoBrowser: DemoBrowser
+
+    override fun onCreate() {
+        super.onCreate()
+        Log.d(TAG, "onCreate")
+        val player = PillarboxPlayer(
+            context = this,
+            mediaItemSource = MixedMediaItemSource(
+                MediaCompositionMediaItemSource(MediaCompositionDataSourceImpl(application, IlHost.PROD))
+            ),
+            /**
+             * If you plan to play some SRG Token protected content
+             */
+            dataSourceFactory = AkamaiTokenDataSource.Factory()
+        )
+        setPlayer(player, DemoCallback())
+
+        demoBrowser = DemoBrowser(DemoPlaylistProvider(this))
+    }
+
+    override fun sessionActivity(): PendingIntent {
+        val intent = Intent(applicationContext, MediaControllerActivity::class.java)
+        val flags = PendingIntentUtils.appendImmutableFlagIfNeeded(PendingIntent.FLAG_UPDATE_CURRENT)
+        return PendingIntent.getActivity(
+            applicationContext,
+            0,
+            intent,
+            flags
+        )
+    }
+
+    override fun onDestroy() {
+        Log.d(TAG, "onDestroy")
+        super.onDestroy()
+    }
+
+    private inner class DemoCallback : MediaLibrarySession.Callback {
+        override fun onGetLibraryRoot(
+            session: MediaLibrarySession,
+            browser: MediaSession.ControllerInfo,
+            params: LibraryParams?
+        ): ListenableFuture<LibraryResult<MediaItem>> {
+            Log.d(TAG, "onGetLibraryRoot")
+            val rootExtras = Bundle().apply {
+                putBoolean(MEDIA_SEARCH_SUPPORTED, false)
+                putBoolean(CONTENT_STYLE_SUPPORTED, true)
+                putInt(CONTENT_STYLE_BROWSABLE_HINT, CONTENT_STYLE_GRID)
+                putInt(CONTENT_STYLE_PLAYABLE_HINT, CONTENT_STYLE_LIST)
+            }
+            val libraryParams = LibraryParams.Builder().setExtras(rootExtras).build()
+            return Futures.immediateFuture(LibraryResult.ofItem(demoBrowser.rootMediaItem, libraryParams))
+        }
+
+        override fun onGetChildren(
+            session: MediaLibrarySession,
+            browser: MediaSession.ControllerInfo,
+            parentId: String,
+            page: Int,
+            pageSize: Int,
+            params: LibraryParams?
+        ): ListenableFuture<LibraryResult<ImmutableList<MediaItem>>> {
+            Log.d(TAG, "onGetChildren($parentId)")
+            return demoBrowser.getChildren(parentId)?.let {
+                Futures.immediateFuture(LibraryResult.ofItemList(it.toImmutableList(), LibraryParams.Builder().build()))
+            } ?: super.onGetChildren(session, browser, parentId, page, pageSize, params)
+        }
+
+        override fun onGetItem(
+            session: MediaLibrarySession,
+            browser: MediaSession.ControllerInfo,
+            mediaId: String
+        ): ListenableFuture<LibraryResult<MediaItem>> {
+            Log.d(TAG, "onGetItem $mediaId")
```

Comment:
⚠️ **[Android Lint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>LogConditional: The log call Log.d(...) should be conditional: surround with `if (Log.isLoggable(...))` or `if (BuildConfig.DEBUG) { ... }`